invision-app/app/components/home/home.js

```diff
@@ -34,29 +31,46 @@ export class Home extends React.Component {
             movies.sort((movieA, movieB) => movieA.release_date > movieB.release_date ? 1 : -1);
         }
 
-        this.setState({
-            movies
-        });
+        setMovies(movies);
+    }
+    function createUpdateMovie(movie) {
+        if (movies.some((m) => m.id != movie.id)) {
+            movies.push(movie);
+            setMovies(movies)
```

Comment:
Add movie functionality does not work because you mutate state. Look at my recommendation fo the **sort** function

Reply:
Modified.